ProfilerGUI/Source/Runner/ProfilerRunner.cs

```diff
@@ -59,16 +55,54 @@ public void RunAsynchronously()
             {
                 logger.Error(e, "Failed to launch application {appPath}", configuration.TargetApplicationPath);
             }
-        }
-
-        private string LocateProfilerDll()
-        {
-            string profilerDll = ProfilerConstants.ProfilerDll64Bit;
-            if (configuration.ApplicationType == EApplicationType.Type32Bit)
+        }
+
+        private static List<(string, string)> DetermineEnvironmentVariables(ProfilerConfiguration configuration)
+        {
+            List<(string, string)> environmentVariables;
+            if (configuration.ApplicationType == EApplicationType.TypeCORE)
+            {
+                environmentVariables = new List<(string, string)>()

```

Comment:
Usually we prefer immediate `return`. You can also initialize the variable dictionary with the variables you add as the last step

Reply:
I split the collection of environment variables into two functions, because they do conceptually different things (hooking up the profiler in the environment vs. configuring the profiler itself). Does this work for you?

Reply:
Sure. reads much better now.